### PR TITLE
add button field; fix #99

### LINKED ISSUE #99: A generic button component with custom functionality
Definitely a "nice-to-have".

I would like to be able to allow the user to have a button within the preferences window that can call functions (ipc back to the parent window process).

Essentially, it would be nice to have a "Reset to defaults" button, or a "Center App Window" button. Some ability to attach custom functionality to a component (to do things like IPC back to the parent window)

### PR DESCRIPTION
**Pull request by lacymorrow**
Adds a button that sends an event to the main process using the `key` property

Use it like:
```js
{
    'label': 'Reset CrossOver Settings',
    'buttonLabel': 'Reset Settings',
    'key': 'resetApp',
    'type': 'button',
    'help': 'Reset all settings to default and clear any custom keybinds',
},
```

And in main process:
```js
prefs.on( 'click', key => {
    if(key === 'resetApp') {
        resetApp()
    }
} )
```

### CODE REVIEW

**Comment by lacymorrow** [comment]
@pvrobays Let me know what you think, this uses a lot of the same code from #29

File: README.md [hunk 1]
```diff
@@ -161,6 +162,14 @@ preferences.on('save', (preferences) => {
 	console.log(`Preferences were saved.`, JSON.stringify(preferences, null, 4));
 });
 
+// Using a button field with `channel: 'reset'`
+preferences.on('click', button => {
```
Comment (pvrobays) [inline README.md:166]: Maybe change the name `button` to `key`. That way it's more clear that the returned string will be the `key` from the preferences.

File: README.md [hunk 2]
```diff
@@ -554,81 +563,89 @@ const preferences = new ElectronPreferences({
...
-                    }
-                ]
-            }
-        },
-        {
-            'id': 'lists',
+							},
+							{
+								'label': 'Ipc button',
+								'key': 'ipcButton',
```
Comment (pvrobays) [inline README.md:573]: In line with the above comment -> maybe call this key `reset`

**Comment by pvrobays** [comment]
@lacymorrow LGTM, nice new feature and well coded. I would only change the documentation a bit to avoid confusion.

**Comment by lacymorrow** [comment]
@pvrobays Good point, totally agree.
Updated the readme to be a little clearer, feel free to kick around for a bit and merge whenever you want. I might go back through #29 again and pull more code out.

I'd love to update dependencies, linters, etc. to make this project a little easier to work with in VSCode, in the future though. Cheers 🍻

**Pull request merged**